hw/xbox/smbus_xbox_smc.c

```diff
@@ -189,6 +220,18 @@ static void smbus_smc_realize(DeviceState *dev, Error **errp)
     if (object_property_get_bool(qdev_get_machine(), "short-animation", NULL)) {
         smc->scratch_reg = SMC_REG_SCRATCH_SHORT_ANIMATION;
     }
+
+    avpack = object_property_get_str(qdev_get_machine(), "avpack", NULL);
+    assert(avpack != NULL);
+    if (strlen(avpack) == 0) {
+        /* pretend to have a composite av pack plugged in by default */
+        smc->avpack_reg = SMC_REG_AVPACK_COMPOSITE;
```

Comment:
Leaving this marker for @mborgerson : default value is hardcoded here + zero-check, but duplicated below.

Is this fine? I consider it bad-code, but don't have a good alternative myself, and we have bikeshedded this a lot already.

@binarymaster said on Discord:

>  `machine_set_avpack` is not called at all if no `avpack` parameter was set in command line
>  but the string value still contains an empty string in that case
>  that's why there are two checks